lib/cloudregister/registerutils.py

```diff
@@ -1398,6 +1404,41 @@ def write_framework_identifier(cfg):
         framework_file.write(json.dumps(identifier))
 
 
+# ----------------------------------------------------------------------------
+def get_cached_instance_data():
+    """Get cached instance data if cache has not expired."""
+    try:
+        with open(get_instance_data_filepath(), 'r') as f:
+            return f.read()
+    except FileNotFoundError:
+        return None
+
+# ----------------------------------------------------------------------------
+def set_instance_data_cache(instance_data):
+    """Update the instance data cache file with new instance data."""
+    with open(get_instance_data_filepath(), 'w') as f:
+        f.write(instance_data)
+
+# ----------------------------------------------------------------------------
+def clear_instance_data_cache():
+    """Remove instance data cache file."""
+    try:
+        os.unlink(get_instance_data_filepath())
```

Comment:
Should we clear all the files with the cache filename format?

Reply:
There is only one instance data cache file per invocation